neo/Consensus/ConsensusContext.cs

```diff
@@ -30,12 +30,18 @@ internal class ConsensusContext : IConsensusContext
         public UInt256[] TransactionHashes { get; set; }
         public Dictionary<UInt256, Transaction> Transactions { get; set; }
         public UInt256[] Preparations { get; set; }
+        public byte[][] PreparationWitnessInvocationScripts { get; set; }
+        public uint[] PreparationTimestamps { get; set; }
         public byte[][] Commits { get; set; }
         public byte[] ExpectedView { get; set; }
+        public byte[][] ChangeViewWitnessInvocationScripts { get; set; }
+        public uint[] ChangeViewTimestamps { get; set; }
+        public byte[] OriginalChangeViewNumbers { get; set; }
```

Comment:
Here is the same, we can combine the four fields `ExpectedView`, `ChangeViewWitnessInvocationScripts`, `ChangeViewTimestamps` and `OriginalChangeViewNumbers`:

```c#
public ConsensusPayload[] ChangeViewPayloads { get; set; }
```

Reply:
Hmm, yes we could do that, but it makes creating the recovery message more work, because we don't need everything in those objects in the recovery message, and also the actual message object inside the `ConsensusPayload` is already serialized and has to be deserialized to set some of the fields when creating the `RecoveryMessage` from it. I don't think it is worth making this change.

Reply:
There are complications serializing for when the object is null.  `Write()` and `ReadSerializableArray` don't handle null objects. So I guess with this change I also have to write additional utility methods for serializing/deserializing an array of Serializable objects that can have nulls.

Reply:
Serialization wasn't that bad, but there is a problem. The `OriginalChangeViewNumbers` are already serialized inside the `ChangeView` message contained in the `ConsensusPayload`, and would have to be deserialized each time when generating the `RecoveryMessage`.

Reply:
I saved off my diff with an attempt to do this, but it has the problem from my last comment, and I stopped. If we are going to try to restructure things to do this, it should be a separate PR after.

Reply:
Are you suggesting the `RecoveryMessage` also has the array of `ConsensusPayload` instead?

Reply:
It has extra redundant information we don't need.

Reply:
I am abandoning this. If you want to do it. How about you give it a shot in a separate PR?

Reply:
> Are you suggesting the `RecoveryMessage` also has the array of `ConsensusPayload` instead?

No. Only `ConsensusContext`.

Reply:
> No. Only `ConsensusContext`.

Then it has the problem I mention above about needing to deserialize the payload each time to be able to generate the recovery message.

`OriginalChangeViewNumbers = ChangeViewPayloads?.Select(p => p.???????),`

Reply:
Can we merge as is? Maybe we could make some new class to combine the fields, but I don't see why it needs to be on this PR.

Reply:
I just read your comment on the other conversation. I'll give it another shot with caching the message in the payload.

Reply:
Implemented the suggestion. Thank you!